pkg/scheduler/scheduler.go

```diff
@@ -291,6 +291,11 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 		cq := snap.ClusterQueues[w.ClusterQueue]
 		ns := corev1.Namespace{}
 		e := entry{Info: w}
+		// the workload cannot be nominated for admission if queuingPolicy is set to Never
+		if workload.IsQueueingPolicyNever(w.Obj) {
```

Comment:
I don't think this is the right place to skip the element.

If this a queue with StrictFIFO, the workload will go back to the head of the queue, and it will block any other workload behind it.

Instead, the workload needs to go outside of the queue, probably by not being in the queue entirely.
I think you could handle this in the event handler in `pkg/controller/core` (methods `WorkloadReconciler.Update` and `WorkloadReconciler.Create`).